src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentDefinitions.java

```diff
@@ -217,4 +182,36 @@
          */
         CANONICAL
     }
+
+    public enum DataSourceType {
+        SIMPLE_XSV("simpleXSV"),
+        LOCATABLE_XSV("locatableXSV"),
+        GENCODE("gencode"),
+        COSMIC("cosmic");
+
+        private final String serialized;
+
+        DataSourceType(final String serializedValue) {
+            serialized = serializedValue;
+        }
+
+        @Override
+        public String toString() {
+            return serialized;
+        }
+
+        public static DataSourceType getEnum(final String s) {
+            for( final DataSourceType val : values() ) {
+                if(val.serialized.equalsIgnoreCase(s)) {
+                    return val;
+                }
+            }
+            throw new IllegalArgumentException("Unexpected value: " + s);
+        }
+    }
+
+    public enum ReferenceVersionType {
```

Comment:
Why do you need this?  Is this just to map the given reference to an internal key?

Reply:
Basically - it's ultimately used to determine which folder in which to search for the data source config files.